coap-gateway/service/config.go

```diff
@@ -16,17 +17,22 @@ import (
 
 //Config represent application configuration
 type Config struct {
-	Log       LogConfig     `yaml:"log" json:"log"`
-	APIs      APIsConfig    `yaml:"apis" json:"apis"`
-	Clients   ClientsConfig `yaml:"clients" json:"clients"`
-	TaskQueue queue.Config  `yaml:"taskQueue" json:"taskQueue"`
+	Log               LogConfig       `yaml:"log" json:"log"`
+	APIs              APIsConfig      `yaml:"apis" json:"apis"`
+	OAuthDeviceClient provider.Config `yaml:"oauthDeviceClient" json:"oauthDeviceClient"`
```

Comment:
DeviceClient naming doesn't make sense anymore. CoAP gateway handles device connections and logic from the authorization server which was more generic was moved directly to the CoAP gateway. The reasoning for this is that its the CoAP which doesn't suppor OAuth protocol; (or, there is no OAuth Server which would support CoAP). OAuth is defined only for HTTP, so this translation from CoAP OAuth `auth code exchange` to HTTP OAuth `auth code exchange` needs to be done in the CoAP Gateway. 

CoAP is API of the CoAP gateway, and if we want to follow the same convention across all the gateways, this configuration (provider.config) should be part of the apis.coap under authorization. -> apis.coap.authorization, same as you have e.g. for the gRPC.

```yaml
apis:
  coap:
    address: "0.0.0.0:5684"
    externalAddress: ""
    maxMessageSize: 262144
    goroutineSocketHeartbeat: "4s"
    keepAlive:
      timeout: 20s
    blockwiseTransfer:
      enabled: true
      blockSize: "1024"
    tls:
      enabled: true
      caPool: "/secrets/public/rootca.crt"
       keyFile: "/secrets/private/cert.key"
       certFile: "/secrets/public/cert.crt"
       clientCertificateRequired: true
    authorization:
        provider: "plgd"
        clientID: ""
        clientSecret: ""
        scopes: []
        authorizationURL: ""
        tokenURL: ""
        redirectURL: ""
        http:
             maxIdleConns: 16
...
```

Reply:
OK, I think it makes sense.